llama_index/indices/vector_store/retrievers/retriever.py

```diff
@@ -121,17 +121,13 @@ def _build_node_list_from_query_result(
             # NOTE: vector store keeps text, returns nodes.
             # Only need to recover image or index nodes from docstore
             for i in range(len(query_result.nodes)):
-                source_node = query_result.nodes[i].source_node
                 if (not self._vector_store.stores_text) or (
-                    source_node is not None and source_node.node_type != ObjectType.TEXT
+                    isinstance(query_result.nodes[i], (ImageNode, IndexNode))
```

Comment:
actually, the query result will always be a TextNode no? -- we check the source type so that we fetch the proper node from the docstore, since vector stores can only store text nodes

Reply:
Just curious what the original node_type was that was causing this check to fail before this change?

Reply:
Maybe we need to double check that we are setting `node_type` properly in the codebase

Reply:
ah yea, looks like this attribute is not being set properly anymore

Reply:
Part of the solution is here:
https://github.com/jerryjliu/llama_index/blob/f55f411d6d5fcc80781215421bc3e4a7ca7238a6/llama_index/vector_stores/utils.py#L46

Rather than dumping the entire node, we have to convert it to a text node and set something to indicate that is was originally not a text node 😓

Reply:
Could shoe-horn the type into the metadata, or add a new attribute to the node class 🤔